Codetest

jvallier opened the pull request:
There we go - realized I made a pull request on my master previously.

jvallier committed:
Add files via upload

jvallier committed:
Merge pull request #1 from jvallier/jvallier-patch-1

Add files via upload

Luwdo commented:
Alright I looked it over and there are a few changes I would like you to make before I accept your PR.

Here the list of issues I noticed, go head and fix them and update your PR.

-In addArmor.php there syntax error on lines 9, 11, and 15, you are missing your closing parentheses on your die calls. Hint: if you follow PSR-2 coding standards it will be easier to spot miss matching parentheses because your brackets will be in different lines.

-For requirement 3 in addArmor.php on line 10, when the second condition of the if statement is reached then 'value' in the post global will not be set (because with && the first condition of the if must be truthy for the second condition to be evaluated) therefore you will not be able to check to see if 'value' is greater then 0 because 'value' is guaranteed to be unset . In addition to changing the && you will need to swap the > to a < to accomplish making sure that we only hit the invalid response if the value is less then 0.

-For requirement 4, you will need to check if an armor exists with the same name and the same type, (currently you are only to checking if an armor of the same name exists) for this you will need to be able to get inventory by name and type, similarly to the way you have changed removals to work. In addition it would be prudent to also check if the type that was entered is in the Armor::getTypeLabels() array.

jvallier committed:
Add files via upload

Some quick fixes and changes based off of the feedback.

Kind of surprised I missed the problem with && and > when looking over it, easily fixed though.

Going to check a few more things on this tomorrow and over the weekend, but in the meantime this should give you an idea of what I'm working towards.

Thank you for your time,
Jon V.

jvallier commented:
This may be a double notification, but I don't know if pushing something to my master/the commits of this pull request sends you a notification that something has been updated or not. So going to duplicate my comments from yesterday just in case:

Some quick fixes and changes based off of the feedback.

Kind of surprised I missed the problem with && and > when looking over it, easily fixed though.

Going to check a few more things on this tomorrow and over the weekend, but in the meantime this should give you an idea of what I'm working towards.

Thank you for your time,
Jon V.

jvallier commented:
Ah hello,
Sorry to be a bother, but did you receive any update e-mails from github?
I added some new commits a few days ago, but I haven't heard anything back yet.
I'm still figuring out some of the workings of github and how it notifies people, so pardon the extra e-mails.
Signed,Jon V. 

    On Tuesday, June 21, 2016 10:59 AM, Nathan Erickson <notifications@github.com> wrote:
 

 Alright I looked it over and there are a few changes I would like you to make before I accept your PR.Here the list of issues I noticed, go head and fix them and update your PR.-In addArmor.php there syntax error on lines 9, 11, and 15, you are missing your closing parentheses on your die calls. Hint: if you follow PSR-2 coding standards it will be easier to spot miss matching parentheses because your brackets will be in different lines.-For requirement 3 in addArmor.php on line 10, when the second condition of the if statement is reached then 'value' in the post global will not be set (because with && the first condition of the if must be truthy for the second condition to be evaluated) therefore you will not be able to check to see if 'value' is greater then 0 because 'value' is guaranteed to be unset . In addition to changing the && you will need to swap the > to a < to accomplish making sure that we only hit the invalid response if the value is less then 0.-For requirement 4, you will need to check if an armor exists with the same name and the same type, (currently you are only to checking if an armor of the same name exists) for this you will need to be able to get inventory by name and type, similarly to the way you have changed removals to work. In addition it would be prudent to also check if the type that was entered is in the Armor::getTypeLabels() array.—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

Luwdo commented:
I took a look and there are still a few issues.

1. You have a missing parentheses on line 14 in removeArmor.php.

2. For the validation check on line 13 in addArmor.php, value doesn't need to be restricted to 1-5 the only invalid armor value would be less then 0.

3. For the validation on line 20 in addArmor.php you are first checking if there is inventory with a specific name then checking if there is an inventory of a specific type. Even though it is is a single expression with a && operator, the checks are still being calculated independent of each other. In order to make sure that there is no other armor with the same name and type as what you are tying to add you would have to do a single check with both criteria being filtered for (1 iteration of the inventory list within a function).

jvallier committed:
Add files via upload

jvallier commented:
Ok, made some changes that should hopefully fix the problems.

Sorry if I seem a little antsy, with the school's open house and Fourth of July plans right around the corner I've been kept on my toes recently.

jvallier commented:
Hello,
I realized I forgot to ask you in the last e-mail the best way to get in contact with you for the GitHub changes.
Because I noticed that you actually can turn off the e-mail notifications GitHub usually sends, so I actually don't know what, if any, notifications you get.
Thank you for your time,Jonathan V.  

    On Thursday, June 30, 2016 2:57 PM, Nathan Erickson <notifications@github.com> wrote:
 

 I took a look and there are still a few issues.   
   - You have a missing parentheses on line 14 in removeArmor.php.
   - For the validation check on line 13 in addArmor.php, value doesn't need to be restricted to 1-5 the only invalid armor value would be less then 0.
   - For the validation on line 20 in addArmor.php you are first checking if there is inventory with a specific name then checking if there is an inventory of a specific type. Even though it is is a single expression with a && operator, the checks are still being calculated independent of each other. In order to make sure that there is no other armor with the same name and type as what you are tying to add you would have to do a single check with both criteria being filtered for (1 iteration of the inventory list within a function).
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

jvallier commented:
Hello,
Just wanted to check in and see if you received my previous e-mail/comment on GitHub.

jvallier committed:
Add files via upload

jvallier commented:
Hello, the initial run problems should be fixed, I ran it in a test server and I no longer got the error over the variable around lines 20 and 40.
